go/pkg/client/bytestream.go

```diff
@@ -26,33 +48,56 @@ func (c *Client) WriteBytes(ctx context.Context, name string, data []byte) error
 	return err
 }
 
+// WriteBytesWithOffset uploads a byte slice with a given resource name to the CAS
+// at an arbitrary offset.
+func (c *Client) WriteBytesWithOffset(ctx context.Context, name string, data []byte, opts ...ByteStreamWriteOption) (int64, error) {
+	ue := uploadinfo.EntryFromBlob(data)
+	ch, err := chunker.New(ue, false, int(c.ChunkMaxSize))
+	if err != nil {
+		return 0, err
+	}
+	return c.writeChunked(ctx, name, ch, opts...)
+}
+
 // writeChunked uploads chunked data with a given resource name to the CAS.
-func (c *Client) writeChunked(ctx context.Context, name string, ch *chunker.Chunker) (int64, error) {
+func (c *Client) writeChunked(ctx context.Context, name string, ch *chunker.Chunker, rawOpts ...ByteStreamWriteOption) (int64, error) {
+	if len(rawOpts) == 1 {
+		return 0, errors.Errorf("Missing an arugment")
+	}
+	opts := &ByteStreamWriteOpts{}
```

Comment:
The `FinishWrite` option should default to `true` to maintain current semantics. That is, we'll always finish at the end of the stream unless instructed otherwise.

Reply:
Done.